### PR TITLE
BUG: Series dot product __rmatmul__ doesn't allow matrix vector multiplication

### LINKED ISSUE #21530: BUG: np.array @ pd.Series swaps arguments
#### Code Sample, a copy-pastable example if possible

Input:
```python
np.zeros((1, 2)) @ pd.Series(np.zeros(2))
```

Output:
```python
  File "<ipython-input-83-3148232e4345>", line 1, in <module>
    np.zeros((1, 2)) @ pd.Series(np.zeros(2))
  File "pandas/core/series.py", line 2061, in __rmatmul__
    return self.dot(other)
  File "pandas/core/series.py", line 2043, in dot
    (lvals.shape, rvals.shape))
Exception: Dot product shape mismatch, (2,) vs (1, 2)
```

#### Problem description

Pandas v0.23.0 (#19035) adds support for expressions `np.array @ pd.Series`, but the support is implemented incorrectly, and instead evaluates `pd.Series @ np.array`, which is different. This is inconsistent with the behaviour of `@` when applied to two arrays, two pandas objects, or an array and a DataFrame. 

#### Expected Output

```python
array([ 0.])
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Linux
OS-release: 4.14.14-200.fc26.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C
LANG: C
LOCALE: None.None

pandas: 0.23.1
pytest: 3.6.0
pip: 10.0.1
setuptools: 39.2.0
Cython: 0.28.3
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 4.2.1
sphinx: None
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.2
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.8
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by minggli**
- [x] closes #21530 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

the new `__rmatmul__` implementation in Series seems to be missing matrix vector multiplication case as raised in the issue in question. Only inner product of two vectors was supported in `__rmatmul__` method.

This PR uses the same implementation in DataFrame to add support in Series and test case.

matmul operator (`@`, `@=`) was added in python 3.5 in https://www.python.org/dev/peps/pep-0465/

```python
class A(object):
    def __matmul__(self, other):
        print('__matmul__ is called in A.')

class B(object):
    def __rmatmul__(self, other):
        print('__rmatmul__ is called in B.')

A() @ B()
B() @ A()
del A.__matmul__
A() @ B()

>>> A() @ B()
__matmul__ is called in A.
>>> B() @ A()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for @: 'B' and 'A'
>>> del A.__matmul__
>>> A() @ B()
__rmatmul__ is called in B.
```

### CODE REVIEW

**Comment by gfyoung** [comment]
@minggli : Looks like a good start!  Can we also add a `whatsnew` too?

File: pandas/tests/series/test_analytics.py [hunk 1]
```diff
@@ -854,6 +854,11 @@ def test_matmul(self):
         expected = np.dot(a.values, a.values)
         assert_almost_equal(result, expected)
 
+        # np.array (matrix) @ Series (__rmatmul__)
```
Comment (gfyoung) [inline pandas/tests/series/test_analytics.py]: Can you reference the issue number too.

**Comment by minggli** [comment]
raised an issue https://github.com/pandas-dev/pandas/issues/21581 regarding error message when `__rmatmul__` is called in dataframe and series. as dot method is implemented treating dataframe or series as left. when `__rmatmul__` is called the error message doesn't work well.

need advice. fix it in the same PR and how to do that or separately.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_analytics.py [hunk 2]
```diff
@@ -849,11 +849,18 @@ def test_matmul(self):
         expected = np.dot(a.values, a.values)
         assert_almost_equal(result, expected)
 
-        # np.array @ Series (__rmatmul__)
-        result = operator.matmul(a.values, a)
+        # GH 21530
+        # vector (list) @ Series (__rmatmul__)
+        result = operator.matmul(a.values.tolist(), a)
```
[the remark this answers is not in the document]
Comment (minggli) [reply pandas/tests/series/test_analytics.py]: np.array @ Series should call `__rmatmul__`, so either list or np.array should be the same, i.e. calling `__rmatmul__`
np.array @ DataFrame however only calls `np.array.__matmul__`

actioned.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @minggli very nice

**Comment by minggli** [comment]
Thanks @jreback and @gfyoung

**Pull request merged**